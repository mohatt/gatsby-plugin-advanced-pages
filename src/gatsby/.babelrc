{
  "presets": [
    ["babel-preset-gatsby-package", { "nodeVersion": "8" }]
  ]
}

