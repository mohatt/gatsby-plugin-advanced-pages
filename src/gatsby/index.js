export { default as onPreBootstrap } from './bootstrap'
export { default as sourceNodes } from './source-nodes'
export { default as setFieldsOnGraphQLNodeType } from './extend-nodes'
export { default as onCreateNode } from './on-create-node'
export { default as createPages } from './create-pages'
export { default as onPreExtractQueries } from './install-fragments'
