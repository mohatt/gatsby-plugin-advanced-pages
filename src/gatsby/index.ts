export type * from './types'
export type { RouteParams, PathGenerator } from '../lib/route-compiler'
