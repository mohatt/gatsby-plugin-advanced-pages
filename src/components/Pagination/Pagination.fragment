import { graphql } from "gatsby"

export const pagination = graphql`
  fragment Pagination on PageInfo {
    perPage
    pageCount
    itemCount
    currentPage
    hasNextPage
    hasPreviousPage
  }
`
