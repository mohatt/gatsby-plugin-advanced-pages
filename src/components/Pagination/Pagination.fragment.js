import { graphql } from 'gatsby'

export const Pagination = graphql`
  fragment Pagination on PageInfo {
    perPage
    pageCount
    itemCount
    currentPage
    hasNextPage
    hasPreviousPage
  }
`
