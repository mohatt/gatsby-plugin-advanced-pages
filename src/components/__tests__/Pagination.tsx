// @vitest-environment jsdom
import React from 'react'
import ReactTest from '@testing-library/react'
import _ from 'lodash'
import { graphql } from 'gatsby'
import { expect, Mock } from 'vitest'
import type { DeepPartial } from '@/types'
import defaultTestCase, { testCases } from './__fixtures__/pagination'
import { Pagination, PaginationProps } from '../Pagination'
import { Pagination as PaginationFragment } from '../Pagination.fragment'

vi.mock('../Link', () => ({
  Link: (props: any) => <a>{JSON.stringify(props)}</a>,
}))

describe('<Pagination />', () => {
  const render = (props?: DeepPartial<PaginationProps>) => {
    const fullProps = _.merge({}, defaultTestCase, props)
    return ReactTest.render(<Pagination {...fullProps} />).container.firstChild
  }

  it('renders correctly with defaults', () => {
    expect(render()).toMatchSnapshot()
  })

  it('renders correctly with custom theme', () => {
    expect(
      render({
        theme: _.mapValues(Pagination.defaultProps.theme, (v) => v + '-custom'),
      }),
    ).toMatchSnapshot()
  })

  it('renders correctly with custom labels', () => {
    expect(
      render({
        labels: _.mapValues(Pagination.defaultProps.labels, (v) => v + ' custom'),
      }),
    ).toMatchSnapshot()
  })

  it('renders correctly with mini ui', () => {
    expect(render({ ui: 'mini' })).toMatchSnapshot()
  })

  it('renders correctly with simple ui', () => {
    expect(render({ ui: 'simple' })).toMatchSnapshot()
  })

  it('renders correctly with full ui', () => {
    expect(render({ ui: 'full' })).toMatchSnapshot()
  })

  it('renders correctly with custom range', () => {
    expect(render({ range: 1 })).toMatchSnapshot()
    expect(render({ range: 3 })).toMatchSnapshot()
    expect(render({ range: 7 })).toMatchSnapshot()
    expect(render({ range: 9 })).toMatchSnapshot()
  })

  it("shouldn't render disabled nav items when renderDisabled is false", () => {
    expect(
      render({
        pageInfo: { currentPage: 1, hasPreviousPage: false },
        renderDisabled: false,
      }),
    ).toMatchSnapshot()
    expect(
      render({
        pageInfo: { currentPage: defaultTestCase.pageInfo.pageCount, hasNextPage: false },
        renderDisabled: false,
      }),
    ).toMatchSnapshot()
    expect(
      render({
        pageInfo: {
          itemCount: 1,
          perPage: 10,
          pageCount: 1,
          currentPage: 1,
          hasNextPage: false,
          hasPreviousPage: false,
        },
        renderDisabled: false,
      }),
    ).toMatchSnapshot()
  })

  it.each(testCases)('correctly handles pageInfo options ($id)', ({ props }) => {
    expect(render(props)).toMatchSnapshot()
  })

  it('correctly defines pagination fragment', () => {
    expect(PaginationFragment).toBeUndefined()
    expect((graphql as Mock).mock.calls).toMatchSnapshot()
  })
})
