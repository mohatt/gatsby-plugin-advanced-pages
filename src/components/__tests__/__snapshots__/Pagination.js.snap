// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should be able to handle different pageInfos 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first disabled"
    >
      <a
        className="page-link"
      >
        « First
      </a>
    </li>
    <li
      className="page-item page-item__prev disabled"
    >
      <a
        className="page-link"
      >
        ← Previous
      </a>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item page-item__next disabled"
    >
      <a
        className="page-link"
      >
        Next →
      </a>
    </li>
    <li
      className="page-item page-item__last disabled"
    >
      <a
        className="page-link"
      >
        Last »
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should be able to handle different pageInfos 2`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item page-item__next disabled"
    >
      <a
        className="page-link"
      >
        Next →
      </a>
    </li>
    <li
      className="page-item page-item__last disabled"
    >
      <a
        className="page-link"
      >
        Last »
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should be able to handle different pageInfos 3`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 7,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 7,
          }
        }
        scope="pagination"
        to="whatever"
      >
        7
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 8,
          }
        }
        scope="pagination"
        to="whatever"
      >
        8
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        9
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 10,
          }
        }
        scope="pagination"
        to="whatever"
      >
        10
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 11,
          }
        }
        scope="pagination"
        to="whatever"
      >
        11
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 12,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom labels 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First custom
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous custom
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next → custom
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last » custom
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom range 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom range 2`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom range 3`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 7,
          }
        }
        scope="pagination"
        to="whatever"
      >
        7
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom range 4`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 7,
          }
        }
        scope="pagination"
        to="whatever"
      >
        7
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 8,
          }
        }
        scope="pagination"
        to="whatever"
      >
        8
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        9
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with custom theme 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination-custom"
  >
    <li
      className="page-item-custom page-item__first-custom"
    >
      <Link
        className="page-link-custom"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item-custom page-item__prev-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item-custom"
    >
      <Link
        className="page-link-custom"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item-custom active-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item-custom page-item__next-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item-custom page-item__last-custom"
    >
      <Link
        className="page-link-custom"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with defaults 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with full ui 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with mini ui 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> should render correctly with simple ui 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> shouldn't render disabled nav items when renderDisabled is false 1`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        2
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 3,
          }
        }
        scope="pagination"
        to="whatever"
      >
        3
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item page-item__next"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 2,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Next →
      </Link>
    </li>
    <li
      className="page-item page-item__last"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        Last »
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> shouldn't render disabled nav items when renderDisabled is false 2`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item page-item__first"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        « First
      </Link>
    </li>
    <li
      className="page-item page-item__prev"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 8,
          }
        }
        scope="pagination"
        to="whatever"
      >
        ← Previous
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 4,
          }
        }
        scope="pagination"
        to="whatever"
      >
        4
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 5,
          }
        }
        scope="pagination"
        to="whatever"
      >
        5
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 6,
          }
        }
        scope="pagination"
        to="whatever"
      >
        6
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 7,
          }
        }
        scope="pagination"
        to="whatever"
      >
        7
      </Link>
    </li>
    <li
      className="page-item"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 8,
          }
        }
        scope="pagination"
        to="whatever"
      >
        8
      </Link>
    </li>
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        params={
          {
            "page": 9,
          }
        }
        scope="pagination"
        to="whatever"
      >
        9
      </Link>
    </li>
  </ul>
</nav>
`;

exports[`<Pagination /> shouldn't render disabled nav items when renderDisabled is false 3`] = `
<nav
  aria-label="Pagination Navigation"
  role="navigation"
>
  <ul
    className="pagination"
  >
    <li
      className="page-item active"
    >
      <Link
        className="page-link"
        scope={null}
        to="whatever"
      >
        1
      </Link>
    </li>
  </ul>
</nav>
`;
